pkg/identificationtype/client_test.go

```diff
@@ -0,0 +1,103 @@
+package identificationtype
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/mercadopago/sdk-go/pkg/config"
+	"github.com/mercadopago/sdk-go/pkg/internal/httpclient"
+)
+
+var (
+	listResponseJSON, _ = os.Open("../../resources/mocks/identification_type/list_response.json")
+	listResponse, _     = io.ReadAll(listResponseJSON)
+)
+
+func TestList(t *testing.T) {
+	type fields struct {
+		config *config.Config
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []Response
+		wantErr string
+	}{
+		{
+			name: "should_return_error_when_creating_request",
+			fields: fields{
+				config: nil,
+			},
+			args: args{
+				ctx: nil,
+			},
+			want:    nil,
+			wantErr: "error creating request: net/http: nil Context",
+		},
+		{
+			name: "should_return_identification_types",
+			fields: fields{
+				config: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							stringReader := strings.NewReader(string(listResponse))
+							stringReadCloser := io.NopCloser(stringReader)
+							return &http.Response{
+								Body: stringReadCloser,
+							}, nil
+						},
+					},
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+			},
+			want: []Response{
+				{
+					ID:        "CPF",
+					Name:      "CPF",
+					Type:      "number",
+					MinLength: 11,
+					MaxLength: 11,
+				},
+				{
+					ID:        "CNPJ",
+					Name:      "CNPJ",
+					Type:      "number",
+					MinLength: 14,
+					MaxLength: 14,
+				},
+			},
+			wantErr: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &client{
+				cfg: tt.fields.config,
+			}
+			got, err := c.List(tt.args.ctx)
+			gotErr := ""
+			if err != nil {
+				gotErr = err.Error()
+			}
+
+			if gotErr != tt.wantErr {
+				t.Errorf("List() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
remover este return

Reply:
Removido.

Reply:
Se remover não vai parar e vai testar o proximo if...

Reply:
`t.Errorf` não interrompe a execução do teste? Se não, devemos manter o `return` mesmo. Sabem @gdeandradero @eltinMeli ?

Reply:
Não interrompe, eu adicionei no meu pq estava testando um erro e para nao criar um mock de sucesso num retorno de erro fiz o return

Reply:
Pelo que vi nos outros testes não estamos utilizando o `return`...

Reply:
mas tem que testar, sao independentes
se o de cima falhou também me interessa ver o de baixo

Reply:
Mas nesse caso, ou passam as duas validação ou não passa no teste, né? Então se cair pra primeira nem precisa testar a segunda mesmo... Certo?

Reply:
certo, mas se a segunda for falhar, que falhe de uma vez
é melhor executar, ver tudo que tem de erro e já corrigir do que corrigir o retorno do erro, rodar o teste e ver que falhou o retorno do valor também e depois corrigir o retorno do valor

Reply:
Pega o cenario que quer testar um retorno do erro, não deveria testar o retorno da api...

Reply:
por que nao?
se eu quero que retorne um erro, nao quero que venha resposta
é certo validar que é nil

Reply:
é só deixar acontecer o nil pointer, o teste já ia falhar de qualquer jeito

Reply:
Deixei o `return`, mas acho que como alguns clients estão sendo testados de ambas as maneiras (com e sem esse return), podemos discutir async e fazer um PR separado padronizando tudo.

Reply:
@gdeandradero sim poderia deixar, o ponto é que não passaria na pipeline do Link aqui no github